src/openforms/registrations/contrib/zgw_apis/typing.py

```diff
@@ -37,6 +39,7 @@ class RegistrationOptions(TypedDict):
     organisatie_rsin: NotRequired[str]
     zaak_vertrouwelijkheidaanduiding: NotRequired[VertrouwelijkheidAanduiding]
     medewerker_roltype: NotRequired[str]
+    objects_api_group: NotRequired[ObjectsAPIGroupConfig]
```

Comment:
```suggestion
    objects_api_group: ObjectsAPIGroupConfig | None
```
I think we can enforce this key being set at all times. If the value is `None`, then no objects API registration is enabled (and we expect `objecttype` and `objecttype_version` to not be set at all, but either way they're ignored)